crates/bevy-inspector-egui/src/reflect_inspector/mod.rs

```diff
@@ -1028,6 +1071,304 @@ impl InspectorUi<'_, '_> {
+                    ui.end_row();
+                });
+
+                if i != len - 1 {
+                    ui.separator();
+                }
+            }
+            let Some(TypeInfo::Set(set_info)) = set.get_represented_type_info() else {
+                return;
+            };
+            let value_type = set_info.value_ty();
+            let (new_op, new_changed) =
+                self.ui_to_insert_set_element_with_options(value_type, ui, id, options);
+            if new_op.is_some() {
+                op = new_op;
+            }
+            changed |= new_changed;
+
+            ui.end_row();
+
+            let error_id = id.with("error");
+
+            // Respond to control interaction
+            if let Some(op) = op {
+                changed |= self.respond_to_set_op(set, &op);
+            }
+
+            let error = ui.data_mut(|data| *data.get_temp_mut_or_default::<bool>(error_id));
+            if error {
+                errors::no_default_value(ui, set_info.type_path());
+            }
+            if ui.input(|input| input.pointer.any_down()) {
+                ui.data_mut(|data| data.insert_temp::<bool>(error_id, false));
+            }
+        });
+
+        changed
+    }
+
+    #[must_use]
+    fn ui_to_insert_set_element_with_options<'a>(
+        &mut self,
+        value_type: bevy_reflect::Type,
+        ui: &mut egui::Ui,
+        id: egui::Id,
+        options: &dyn Any,
+    ) -> (Option<SetOp>, bool) {
+        let mut changed = false;
+        let mut op = None;
+        ui.vertical(|ui| {
+            ui.label("New element");
+            let set_draft_id = id.with("set_draft");
+            let draft_clone = ui.data_mut(|data| {
+                data.get_temp_mut_or_default::<Option<SetDraftElement>>(set_draft_id)
+                    .to_owned()
+            });
+            ui.end_row();
+            match draft_clone {
+                None => {
+                    // If no draft element exists, show a button to create one.
+                    if add_button(ui).clicked() {
+                        // Insert a temporary 'draft' value into UI state, once inserted, we cannot modify it.
+                        let maybe_default = self
+                            .get_default_value_for(value_type.id())
+                            .map(|v| SetDraftElement(v.into_partial_reflect()));
+                        if maybe_default.is_some() {
+                            ui.data_mut(|data| data.insert_temp(set_draft_id, maybe_default));
+                        }
+                    }
+                    ui.end_row();
+                }
+                Some(SetDraftElement(mut v)) => {
+                    ui.end_row();
+                    // Show controls for editing our draft element.
+                    // FIXME: is the id passed here correct?
+                    let value_changed =
+                        self.ui_for_reflect_with_options(v.as_mut(), ui, id, options);
```

Comment:
I'm not sure this requires an id ; it doesn´t seem related to the rest of the data as it's a draft, but I'm not sure.